cdap-app-fabric/src/main/java/io/cdap/cdap/internal/capability/CapabilityApplier.java

```diff
@@ -82,26 +95,34 @@
   private static final ProgramTerminator NOOP_PROGRAM_TERMINATOR = programId -> {
     // no-op
   };
+  private static final int DEFAULT_AUTO_INSTALL_THREADS = 5;
   private final SystemProgramManagementService systemProgramManagementService;
   private final ApplicationLifecycleService applicationLifecycleService;
   private final ProgramLifecycleService programLifecycleService;
   private final NamespaceAdmin namespaceAdmin;
   private final CapabilityStatusStore capabilityStatusStore;
   private final MetadataSearchClient metadataSearchClient;
   private final ArtifactRepository artifactRepository;
+  private final File tmpDir;
+  private ExecutorService executorService;
+  private final int autoInstallThreads;
 
   @Inject
   CapabilityApplier(SystemProgramManagementService systemProgramManagementService,
                     ApplicationLifecycleService applicationLifecycleService, NamespaceAdmin namespaceAdmin,
                     ProgramLifecycleService programLifecycleService, CapabilityStatusStore capabilityStatusStore,
-                    ArtifactRepository artifactRepository, DiscoveryServiceClient discoveryClient) {
+                    ArtifactRepository artifactRepository, DiscoveryServiceClient discoveryClient,
+                    CConfiguration cConf) {
     this.systemProgramManagementService = systemProgramManagementService;
     this.applicationLifecycleService = applicationLifecycleService;
     this.programLifecycleService = programLifecycleService;
     this.capabilityStatusStore = capabilityStatusStore;
     this.namespaceAdmin = namespaceAdmin;
     this.metadataSearchClient = new MetadataSearchClient(discoveryClient);
     this.artifactRepository = artifactRepository;
+    this.tmpDir = new File(cConf.get(Constants.CFG_LOCAL_DATA_DIR),
+                           cConf.get(Constants.AppFabric.TEMP_DIR)).getAbsoluteFile();
+    this.autoInstallThreads = cConf.getInt(Constants.Capability.AUTO_INSTALL_THREADS, DEFAULT_AUTO_INSTALL_THREADS);
```

Comment:
default value for configuration should be declared in cdap-default.xml, not in code.

Reply:
fixed